agent/agentcontainers/api.go

```diff
@@ -142,6 +168,56 @@ func (api *API) getContainers(ctx context.Context) (codersdk.WorkspaceAgentListC
 	api.containers = updated
 	api.mtime = now
 
+	api.devcontainersMu.Lock()
+	defer api.devcontainersMu.Unlock()
+
+	// Reset all known devcontainers to not running.
+	for i := range api.knownDevcontainers {
+		api.knownDevcontainers[i].Running = false
+		api.knownDevcontainers[i].Container = nil
+	}
+
+	// Check if the container is running and update the known devcontainers.
+	for _, container := range updated.Containers {
+		workspaceFolder := container.Labels[DevcontainerLocalFolderLabel]
```

Comment:
thought, non-blocking: Given that a `WorkspaceAgentDevcontainer` is a superset of a `WorkspaceAgentContainer`, I wonder if we could instead store the optional devcontainer-specific fields in `WorkspaceAgentContainer`?

Reply:
The main issue with that approach is: how do we represent a devcontainer we know about but that has no container? (I.e. either it wasn't started, or it was deleted.)

Reply:
Gotcha 👍